### PR TITLE
CFY-7197 Propagate network to agent

### PR DESCRIPTION
**Pull request by mcouthon**

### CODE REVIEW

File: cloudify_agent/tests/installer/config/test_configuration.py [hunk 1]
```diff
@@ -125,32 +136,10 @@ def test_prepare_secured(self):
...
-                   'ssl_cert_path': '/tmp/blabla',
-               }}
-           ))
-    def test_connection_params_propagation(self):
-        # Testing that if a connection timeout is passed as an agent runtime
-        # property, it would be propagated to the cloudify agent dict
-        cloudify_agent = CloudifyAgentConfig({'local': True})
-        cloudify_agent.set_initial_values(True)
-        self.assertEqual(cloudify_agent['ssl_cert_path'], '/tmp/blabla')
+        self.assertDictEqual(expected, cloudify_agent)
```
Comment (tehasdf) [inline cloudify_agent/tests/installer/config/test_configuration.py]: hey add a newline at the end of file, this is very important

File: cloudify_agent/operations.py [hunk 2]
```diff
@@ -172,11 +172,13 @@ def _set_default_new_agent_config_values(old_agent, new_agent):
...
     new_agent['disable_requiretty'] = False
     new_agent['install_with_sudo'] = True
+    new_agent['networks'] = ctx.bootstrap_context.cloudify_agent.networks
 
 
 def _copy_values_from_old_agent_config(old_agent, new_agent):
     fields_to_copy = ['windows', 'ip', 'basedir', 'user', 'distro_codename',
-                      'distro', 'broker_ssl_cert_path', 'agent_rest_cert_path']
+                      'distro', 'broker_ssl_cert_path', 'agent_rest_cert_path',
+                      'network']
```
Comment (tehasdf) [inline cloudify_agent/operations.py]: i think `old_agent` might not have "network" yet?

File: cloudify_agent/installer/config/agent_config.py [hunk 3]
```diff
@@ -150,15 +150,19 @@ def _set_name(self):
...
+            manager_ip = networks.get(network)
+            if not manager_ip:
+                raise exceptions.AgentInstallerConfigurationError(
+                    'The network associated with the agent (`{0}`) does not '
+                    'appear in the list of manager networks assigned at '
+                    'bootstrap ({1})'.format(network, ', '.join(networks))
+                )
+        else:
+            # Might be getting here when working in local workflows (or tests)
+            manager_ip = cloudify_utils.get_manager_rest_service_host()
```
Comment (tehasdf) [inline cloudify_agent/installer/config/agent_config.py]: does this actually work with local workflows? that utils getter only looks in the environment, I think this is only relevant to the mgmtworker?

Comment (mcouthon) [reply cloudify_agent/installer/config/agent_config.py]: Well... this is how the tests do it :) I'll see if it's relevant for local workflows

File: cloudify_agent/operations.py (hunk 2 above)
Comment (mcouthon) [reply cloudify_agent/operations.py]: We `if field in old_agent:`, so it should it be fine, no?

File: cloudify_agent/installer/config/agent_config.py (hunk 3 above)
Comment (mcouthon) [reply cloudify_agent/installer/config/agent_config.py]: Well, seeing how this is how it used to work so far, I feel Ok with leaving it here: https://github.com/cloudify-cosmo/cloudify-agent/blob/master/cloudify_agent/installer/config/agent_config.py#L120

File: cloudify_agent/operations.py (hunk 2 above)
Comment (tehasdf) [reply cloudify_agent/operations.py]: oh, we do indeed!

File: cloudify_agent/installer/config/agent_config.py (hunk 3 above)
Comment (tehasdf) [reply cloudify_agent/installer/config/agent_config.py]: cool

**Pull request merged**